zellij-tile/src/data.rs

```diff
@@ -32,6 +32,7 @@ pub enum Event {
     TabUpdate(Vec<TabInfo>),
     KeyPress(Key),
     Timer(f64),
+    InputStringUpdate(String),
```

Comment:
Is there a really vital reason to use something like this instead of parsing key-presses like we do already? The naming of tabs just uses the `KeyPress(Key)` event. How reusable do we think that `InputStringUpdate(String)` will be?

Reply:
This was actually the first approach I tried. It may be because of my unfamiliarity with the API, but I ran into a couple of issues with using `KeyPress` events.. From what I can tell, keyboard input is processed twice as both the raw byte input and the `KeyPress` type. On top of that, the only place `KeyPress` events are currently generated seems to be where writing to the active terminal, which (correct me if I'm wrong) is not called when in an input mode. By the time a keypress reaches the `SearchTab` function, it's already been converted to a `Vec<u8>` which from what I can tell cannot be easily converted to a character with which to send a `KeyPress` event to the plugin.

As far as reusability, I could definitely see generalized InputStrings being used for a number of plugin and application operations, but the only other existing use case would I guess be updateTabName (which writes directly to the tab instead). For what it's worth, I would rather have the display implementation tied to the tab bar but I'm not sure I'm familiar enough with the rendering system to figure out if it's feasible.

Reply:
You raise some good points! How would you feel about grouping the current search string in with a TabUpdate somehow? It's just that subscribing to an "InputString" event isn't super specific. Several plugins could reasonably want updates like that, but it wouldn't be clear what the strings actually represent (a search string in the tab-bar, a file name in a future version of strider).

If it's with the TabUpdate, then the data obviously shows it's related to tabs. What do you think?

Reply:
> an "InputString" event isn't super specific

I think my reasoning behind this was that a user would only be dealing with one input string at any given time and I was only thinking about the display functionality so I didn't think it would matter what the actual purpose of the string would be. That said, I could see the value in having more specific categorizations.

> If it's with the TabUpdate, then the data obviously shows it's related to tabs. What do you think?

This would be a reasonable option, though since the Tab info is currently passed in as a vector it would need to be wrapped in a custom data type.

> Several plugins could reasonably want updates like that, but it wouldn't be clear what the strings actually represent (a search string in the tab-bar, a file name in a future version of strider).

Maybe an alternative to the tab option would be to keep an input string event, but pass in a map or custom data type with additional info for the input string(s) so that their purpose is clear. Either the key could be an input prompt (removing the need to put it in ModeInfo) or some sort of identifier that can be resolved by the plugin. Do you think this is a reasonable/feasible alternative?